ADotNet/Services/Builds/BuildService.Exceptions.cs

```diff
@@ -23,49 +24,75 @@ private static void TryCatch(ReturningNothingFunction returningNothingFunction)
             }
             catch (NullPipelineException nullPipelineException)
             {
-                throw new AdoValidationException(nullPipelineException);
+                throw CreateAdoValidationException(nullPipelineException);
             }
             catch (NullPathException nullPathException)
             {
-                throw new AdoValidationException(nullPathException);
+                throw CreateAdoValidationException(nullPathException);
             }
             catch (ArgumentNullException argumentNullException)
             {
-                throw new AdoDependencyValidationException(argumentNullException);
+                throw CreateAdoDependencyValidationException(argumentNullException);
             }
             catch (ArgumentException argumentException)
             {
-                throw new AdoDependencyValidationException(argumentException);
+                throw CreateAdoDependencyValidationException(argumentException);
             }
             catch (PathTooLongException pathTooLongException)
             {
-                throw new AdoDependencyValidationException(pathTooLongException);
+                throw CreateAdoDependencyValidationException(pathTooLongException);
             }
             catch (DirectoryNotFoundException directoryNotFoundException)
             {
-                throw new AdoDependencyValidationException(
-                    directoryNotFoundException);
+                throw CreateAdoDependencyValidationException(directoryNotFoundException);
             }
             catch (IOException ioException)
             {
-                throw new AdoDependencyException(ioException);
+                throw CreateAdoDependencyException(ioException);
             }
             catch (SecurityException securityException)
             {
-                throw new AdoDependencyException(securityException);
+                throw CreateAdoDependencyException(securityException);
             }
             catch (UnauthorizedAccessException unauthorizedAccessException)
             {
-                throw new AdoDependencyException(unauthorizedAccessException);
+                throw CreateAdoDependencyException(unauthorizedAccessException);
             }
             catch (NotSupportedException notSupportedException)
             {
-                throw new AdoDependencyException(notSupportedException);
+                throw CreateAdoDependencyException(notSupportedException);
             }
             catch (Exception exception)
             {
-                throw new BuildServiceException(exception);
+                throw CreateBuildServiceException(exception);
             }
         }
+
+        private static AdoValidationException CreateAdoValidationException(Xeption innerException)
+        {
+            return new AdoValidationException(
+                "Ado validation exception occurred, try again", innerException);
```

Comment:
use parameters names and one parameter per line if not on the same line as the exception initialization

Reply:
Treated